### PR TITLE
[Carry 623] Fix can't save image by image id

### PR DESCRIPTION
**Pull request by djdongjin**
Carry changes from #623 and also make some refactor to `nerdctl save`.

1. Make `nerdctl save` work when image ids are given; also works if the same image has multiple names (only one copy is saved); return error if an id prefix corresponds to multiple different images.
2. Refactor `nerdctl save`: make `saveImages` not depend on cobra.

### CODE REVIEW

File: cmd/nerdctl/save_linux_test.go [hunk 1]
```diff
@@ -48,6 +48,43 @@ func TestSave(t *testing.T) {
 	assert.Assert(t, strings.Contains(etcOSRelease, "Alpine"))
 }
 
+func TestSaveById(t *testing.T) {
+	base := testutil.NewBase(t)
+	base.Cmd("pull", testutil.AlpineImage).AssertOK()
+	inspect := base.InspectImage(testutil.AlpineImage)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/save_linux_test.go:54]:
```suggestion
	inspect := base.InspectImage(testutil.Commonmage)
```

And the test can be moved to `save_test.go`

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: cmd/nerdctl/save_linux_test.go [hunk 2]
```diff
@@ -33,17 +33,17 @@ import (
...
 	err := extractDockerArchive(archiveTarPath, rootfsPath)
 	assert.NilError(t, err)
 	etcOSReleasePath := filepath.Join(rootfsPath, "/etc/os-release")
 	etcOSReleaseBytes, err := os.ReadFile(etcOSReleasePath)
 	assert.NilError(t, err)
 	etcOSRelease := string(etcOSReleaseBytes)
-	t.Logf("read %q, extracted from %q", etcOSReleasePath, testutil.AlpineImage)
+	t.Logf("read %q, extracted from %q", etcOSReleasePath, testutil.CommonImage)
 	t.Log(etcOSRelease)
 	assert.Assert(t, strings.Contains(etcOSRelease, "Alpine"))
```
Comment (AkihiroSuda) [inline cmd/nerdctl/save_linux_test.go:48]: The test is specific to Alpine, so you can retain `testutil.AlpineImage` for this test

Comment (djdongjin) [reply cmd/nerdctl/save_linux_test.go]: fixed

File: cmd/nerdctl/save_linux_test.go (hunk 1 above)
Comment (djdongjin) [reply cmd/nerdctl/save_linux_test.go]: fixed

**Pull request merged**